lib/google_crawler/accounts/user.ex

```diff
@@ -0,0 +1,35 @@
+defmodule GoogleCrawler.Accounts.User do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "users" do
+    field :email, :string
+    field :username, :string
+    field :encrypted_password, :string
+
+    field :password, :string, virtual: true
+    field :password_confirmation, :string, virtual: true
+
+    timestamps()
+  end
+
+  @fields ~w(email username encrypted_password password password_confirmation)a
+
+  # TODO: Add Password confirmation
+  def changeset(user, attrs) do
+    user
+    |> cast(attrs, @fields)
+    |> validate_required([:email, :username])
+    |> validate_format(:email, ~r/@/)
+    |> validate_length(:password, min: 6)
+    |> unique_constraint(:email)
+    |> unique_constraint(:username)
+    |> encrypt_password
+  end
+
+  defp encrypt_password(%Ecto.Changeset{valid?: true, changes: %{password: password}} = changeset) do
```

Comment:
Not sure if it's worth to pattern match on the validity of the changeset as it requires to create two methods (when it's not valid) 🤔 When inserting in the database, it would already check for a valid changeset. 

I think probably the good part would be to only empty the password when the password is actually changed?

Reply:
I ended up doing something similar minus pattern match on the valid changeset: https://github.com/olivierobert/crawlexir/pull/9/commits/501e6d66f0b1efc197b84330352e9bf06b39d2a5 🤔